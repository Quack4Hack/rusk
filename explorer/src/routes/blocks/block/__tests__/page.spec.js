import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import Block from "../+page.svelte";

global.ResizeObserver = vi.fn().mockImplementation(() => ({
  disconnect: vi.fn(),
  observe: vi.fn(),
  unobserve: vi.fn(),
}));

describe("Block", () => {
  vi.useFakeTimers();
  vi.setSystemTime(new Date(2024, 4, 15));

  afterEach(() => {
    cleanup();
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  it("should render the Block Details page", () => {
    const { container } = render(Block, {});

    expect(container.firstChild).toMatchSnapshot();
  });
});
