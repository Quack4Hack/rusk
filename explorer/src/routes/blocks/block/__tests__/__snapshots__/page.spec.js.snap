// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Block > should render the Block page 1`] = `
<div>
  <section
    class="block s-vKBwXbsZSIaK"
  >
    <div
      class="block__details s-vKBwXbsZSIaK"
    >
      Block details
    </div>
     
    <div
      class="block__transactions s-vKBwXbsZSIaK"
    >
      Block transactions
    </div>
  </section>
  <!--&lt;+page&gt;-->
</div>
`;
